scipy/sparse/construct.py

```diff
@@ -770,10 +768,32 @@ def random(m, n, density=0.01, format='coo', dtype=None,
     elif isinstance(random_state, (int, np.integer)):
         random_state = np.random.RandomState(random_state)
     if data_rvs is None:
-        data_rvs = random_state.rand
+        if np.issubdtype(dtype, np.integer):
+            try:
+                np.lib.NumpyVersion
+            except AttributeError:
```

Comment:
We don't need to support numpy versions this old, so this check can be simplified.

Reply:
Okay, I simplified it.